bert_e/workflow/gitwaterflow/branches.py

```diff
@@ -666,17 +666,36 @@ def queued_prs(self):
         if not self._queues:
             return []
 
+        # Find last_entry for which there is not a hf entry
+        last_entry = None
         pr_ids = []
-        for key in list(self._queues.keys()):
-            entry = self._queues[key]
-            pr_ids = pr_ids + [branch.pr_id for branch in
-                               entry[QueueIntegrationBranch]]
-
-        # remove duplicate entries from pr_ids
-        pr_ids = list(dict.fromkeys(pr_ids))
-
-        pr_ids.sort()
-        return pr_ids
+        for key in list(reversed(self._queues.keys())):
+            if len(key) < 4:
+                last_entry = self._queues[key]
+                break
+
+        if last_entry is not None:
+            pr_ids = list(reversed([branch.pr_id for branch in
+                                    last_entry[QueueIntegrationBranch]]))
+
+        # Add hotfix PRs that are not seen from the queues top key
+        pr_hf_ids = []
+        for key in list(reversed(self._queues.keys())):
+            if len(key) == 4:
+                entry = self._queues[key]
+                new_pr_ids = list([branch.pr_id for branch in
+                                   entry[QueueIntegrationBranch]])
+                for pr_hf_id in new_pr_ids:
+                    if pr_hf_id not in pr_hf_ids:
+                        pr_hf_ids = [pr_hf_id] + pr_hf_ids
+
+        # Remove hotfix PRs from the first set
+        pr_non_hf_ids = []
+        for pr_id in pr_ids:
+            if pr_id not in pr_hf_ids:
+                pr_non_hf_ids = pr_non_hf_ids + [pr_id]
+
+        return pr_hf_ids + pr_non_hf_ids
```

Comment:
how this one works?

Reply:
this was explained offline, we're listing hotfix prs first and then the normal PR